From 2600e318156d96a2539d79db15939a6a73d7021e Mon Sep 17 00:00:00 2001
From: "Damien.Horsley" <Damien.Horsley@imgtec.com>
Date: Thu, 14 Jan 2016 14:58:05 +0000
Subject: ASoC: img: initialise Pistachio active_rate to zero

Initialise active_rate to zero in pistachio_card_change_rate
to correct the behaviour when calling hw_params multiple
times for a stream

Change-Id: I807a05a13817956492a8f3e1f013e5ececb36916
Signed-off-by: Damien.Horsley <Damien.Horsley@imgtec.com>
---
 sound/soc/img/pistachio.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/img/pistachio.c b/sound/soc/img/pistachio.c
index 1a8a6d7..8fa34fb 100644
--- a/sound/soc/img/pistachio.c
+++ b/sound/soc/img/pistachio.c
@@ -455,6 +455,7 @@ static int pistachio_card_change_rate(struct pistachio_card *pbc,
 	int ret;
 
 	mutex_lock(&pbc->rate_mutex);
+	*active_rate = 0;
 	ret = _pistachio_card_change_rate(pbc, rate, i2s);
 	if (!ret)
 		*active_rate = rate;
-- 
1.9.1

